File: Cocoa/AppFrameworks/UIKit/CollectionView/YJUICollectionView.m
//
//  YJUICollectionView.m
//  YJCocoa
//
//  HomePage:https://github.com/937447974/YJCocoa
//  YJ技术支持群:557445088
//
//  Created by 阳君 on 2018/6/9.
//  Copyright © 2016年 YJCocoa. All rights reserved.
//


#import "YJUICollectionView.h"

@implementation YJUICollectionView

- (instancetype)initWithFrame:(CGRect)frame collectionViewLayout:(UICollectionViewLayout *)layout {
    self = [super initWithFrame:frame collectionViewLayout:layout];
    if (self) {
        _manager = [[YJUICollectionViewManager alloc] initWithCollectionView:self];
    }
    return self;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    _manager = [[YJUICollectionViewManager alloc] initWithCollectionView:self];
}

#pragma mark - getter
- (NSMutableArray<YJUICollectionCellObject *> *)dataSourcePlain {
    return self.manager.dataSourcePlain;
}

- (void)setDataSourcePlain:(NSMutableArray<YJUICollectionCellObject *> *)dataSourcePlain {
    self.manager.dataSourcePlain = dataSourcePlain;
}

- (NSMutableArray<NSMutableArray<YJUICollectionCellObject *> *> *)dataSourceGrouped {
    return self.manager.dataSourceGrouped;
}

- (NSMutableArray<YJUICollectionCellObject *> *)dataSourceHeader {
    return self.manager.dataSourceManager.dataSourceHeader;
}

- (NSMutableArray<YJUICollectionCellObject *> *)dataSourceFooter {
    return self.manager.dataSourceManager.dataSourceFooter;
}

@end

File: YJCocoa/Classes/AppFrameworks/UIKit/CollectionView/YJUICollectionView.swift
//
//  YJUICollectionView.swift
//  YJCollectionView
//
//  Created by 阳君 on 2019/5/23.
//  Copyright © 2019 YJCocoa. All rights reserved.
//

import UIKit

@objcMembers
open class YJUICollectionView: UICollectionView {

    /// 管理器
    public lazy var manager: YJUICollectionViewManager! = {
        let _manager = YJUICollectionViewManager(collectionView: self)
        self.delegate = _manager
        self.dataSource = _manager
        return _manager
    }()
    /// header 数据源
    public var dataSourceHeader: Array<YJUICollectionCellObject> {
        get {return self.manager.dataSourceHeader}
        set {self.manager.dataSourceHeader = newValue}
    }
    /// footer 数据源
    public var dataSourceFooter: Array<YJUICollectionCellObject> {
        get {return self.manager.dataSourceFooter}
        set {self.manager.dataSourceFooter = newValue}
    }
    /// cell 数据源
    public var dataSourceCell: Array<Array<YJUICollectionCellObject>> {
        get {return self.manager.dataSourceCell}
        set {self.manager.dataSourceCell = newValue}
    }
    /// cell 第一组数据源
    public var dataSourceCellFirst: Array<YJUICollectionCellObject> {
        get {return self.manager.dataSourceCellFirst}
        set {self.manager.dataSourceCellFirst = newValue}
    }
    
}
